include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,405 @@
+
+namespace detail {
+
+// offset_index_sequence idea comes from "offset_sequence" here:
+// https://devblogs.microsoft.com/oldnewthing/20200625-00/?p=103903
+//
+// offset_index_sequence adds N to each element of the given IndexSequence.
+// We can't just template on the parameter pack of indices directly;
+// the pack needs to be contained in some type.
+// We choose index_sequence because it stores no run-time data.
+template <size_t N, class IndexSequence>
+struct offset_index_sequence;
+
+template <size_t N, size_t... Indices>
+struct offset_index_sequence<N, index_sequence<Indices...>>
+{
+  using type = index_sequence<(Indices + N)...>;
+};
+
+template <size_t N, typename IndexSequence>
+using offset_index_sequence_t = typename offset_index_sequence<N, IndexSequence>::type;
+
+// iota_index_sequence defines the half-open sequence
+// begin, begin+1, begin+2, ..., end-1.
+// If end == begin, then the sequence is empty (we permit this).
+//
+// Defining the struct first, rather than going straight to the type alias,
+// lets us check the template arguments.
+template <size_t begin, size_t end>
+struct iota_index_sequence
+{
+  static_assert(end >= begin, "end must be >= begin.");
+  using type =
+      offset_index_sequence_t<begin, std::make_index_sequence<end - begin>>;
+};
+
+template <class _Extents, class _Enabled = void>
+struct __p_left;
+
+template <class _Extents>
+struct __p_left<_Extents, enable_if_t<(_Extents::rank() < 2)>>
+{
+  using __type = std::index_sequence<>;
+};
+
+template <class _IndexType, size_t... _Extents>
+struct __p_left<extents<_IndexType, _Extents...>>
+{
+  using __extents_type = extents<_IndexType, _Extents...>;
+  using __type = typename iota_index_sequence<1, __extents_type::rank()>::type;
+};
+
+template<class _T>
+MDSPAN_INLINE_FUNCTION
+constexpr auto
+__find_aligned_offset(_T __alignment, _T __offset)
+{
+  if ( __alignment == 0 )
+    return _T(0);
+  else
+    return ( ( __offset + __alignment - 1 ) / __alignment) * __alignment;
+}
+
+template<class _ExtentsType, size_t _PaddingStride>
+MDSPAN_INLINE_FUNCTION
+constexpr size_t
+__get_actual_padding_stride()
+{
+  constexpr auto __rank = _ExtentsType::rank();
+
+  if constexpr (__rank <= size_t(1)) {
+    return _PaddingStride;
+  } else if constexpr (_PaddingStride != dynamic_extent &&
+                       _ExtentsType::static_extent(0) != dynamic_extent) {
+    static_assert((_PaddingStride != 0) || (_ExtentsType::static_extent(0) == 0), "padding stride can be 0 only if extents_type::static_extent(0) is 0");
+    return __find_aligned_offset(_PaddingStride, _ExtentsType::static_extent(0));
+  } else {
+    return dynamic_extent;
+  }
+}
+
+template<size_t _ExtentToSub, class _Extents, size_t _NewExtent, class _Indices>
+struct __substitute_extents_impl;
+
+template<size_t _ExtentToSub, class _Extents, size_t _NewExtent, size_t... _Indices>
+struct __substitute_extents_impl<_ExtentToSub, _Extents, _NewExtent, index_sequence<_Indices...>>
```

Comment:
if you use `extents<IndexType, OrigExtents...>` here instead of `_Extents` (and use those two as template parameters), you can later directly use `IndexType` instead of `typename _Extents::index_type` and `OrigExtents` instead of `_Extents::static_extent(_Indices)`.
You can still make a typedef for places where you use `_Extents` on their own right now.

Reply:
Changed according to your suggestion